.github/actions/nr-upload/index.js

```diff
@@ -0,0 +1,88 @@
+import { v4 as uuidv4 } from 'uuid'
+import { args } from './args.js'
+import { fetchRetry } from '../shared-utils/fetch-retry.js'
+import { constructLoaderFileNames, constructFuzzyVersions } from '../shared-utils/loaders.js'
+
+const loaderFileNames = constructLoaderFileNames(args.loaderVersion)
+const fuzzyVersions = constructFuzzyVersions(args.loaderVersion)
+const envOptions = {
+  stage: {
+    url: 'https://staging-api.newrelic.com/v2/js_agent_loaders/create.json',
+    headers: {
+      'X-Api-Key': args.stageApiKey
+    }
+  },
+  prod: {
+    url: 'https://api.newrelic.com/v2/js_agent_loaders/create.json',
+    headers: {
+      'X-Api-Key': args.prodApiKey
+    }
+  }
+}
+
+const loaderFileContents = (await Promise.all(
+  loaderFileNames.map(async loaderFileName => {
+    const fileContentsRequest = await fetchRetry(`https://js-agent.newrelic.com/${loaderFileName}?_nocache=${uuidv4()}`, { retry: 3 })
+
+    if (!fileContentsRequest.ok) {
+      throw new Error(`Download for loader ${loaderFileName} failed.`)
+    }
+
+    const fileContents = await fileContentsRequest.text()
+
+    return [loaderFileName, fileContents]
+  })
+)).reduce((aggregator, [loaderFileName, fileContents]) => {
+  aggregator[loaderFileName] = fileContents
+  return aggregator
+}, {})
+
+const uploadJobs = args.environment.map(env => {
+  return Object.entries(loaderFileContents).map(([loaderFileName, loaderFileContent]) => {
+    return [
+      {
+        env,
+        loaderFileName,
+        loaderFileContent
+      },
+      {
+        env,
+        loaderFileName: loaderFileName.replace(args.loaderVersion, fuzzyVersions.PATCH),
+        loaderFileContent
+      },
+      {
+        env,
+        loaderFileName: loaderFileName.replace(args.loaderVersion, fuzzyVersions.MINOR),
+        loaderFileContent
+      }
+    ]
+  }).flat()
+}).flat()
+
+await Promise.allSettled(
```

Comment:
Check with API-E hero to see if there is any rate limiting concerns here.

Reply:
The [limit](https://docs.newrelic.com/docs/apis/rest-api-v2/basic-functions/api-overload-protection-handling-429-errors/) is 1000 calls per minute so we should never hit that limit.